src/librustc/session/mod.rs

```diff
@@ -859,32 +861,43 @@ impl Session {
                  self.perf_stats.normalize_projection_ty.load(Ordering::Relaxed));
     }
 
-    /// We want to know if we're allowed to do an optimization for crate foo from -z fuel=foo=n.
-    /// This expends fuel if applicable, and records fuel if applicable.
-    pub fn consider_optimizing<T: Fn() -> String>(&self, crate_name: &str, msg: T) -> bool {
+    #[inline(never)]
+    #[cold]
+    pub fn consider_optimizing_cold<T: Fn() -> String>(&self, crate_name: &str, msg: T) -> bool {
```

Comment:
Does splitting this into two functions really translate to faster compile-times?

Reply:
It is used in layout computation. Not sure how hot it is, but it probably won't hurt. Especially if we want to add more MIR  optimizations (and debug them)

Reply:
Anyway this PR is mostly about getting rid of `LockCell`, I just found this on the way

Reply:
OK. It does hurt code quality though as it makes things more complicated and thus harder to read and maintain. I personally prefer not to do things like this without evidence that it carries it's weight.